vercel/resource_edge_config_item.go

```diff
@@ -0,0 +1,260 @@
+package vercel
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/vercel/terraform-provider-vercel/client"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource              = &edgeConfigItemResource{}
+	_ resource.ResourceWithConfigure = &edgeConfigItemResource{}
+)
+
+type edgeConfigItemResource struct {
+	client *client.Client
+}
```

Comment:
This needs wiring up in the `provider.go` file. We could also do with tests, and given there's a fairly sane `Read` method I think we could add a data source too.

Reply:
I've wired this up and added an equivalent data source.